src/main/java/com/traveltime/sdk/dto/requests/TimeFilterFastProtoRequest.java

```diff
@@ -87,4 +90,35 @@
             .<TravelTimeError>mapLeft(cause -> new IOError(cause, IO_PROTO_ERROR + cause.getMessage()))
             .flatMap(this::parseResponse);
     }
+
+    static public List<TimeFilterFastProtoRequest> split(TimeFilterFastProtoRequest request, int batchSizeHint) {
+        /* Naively splitting requests may lead to situations where the last request is very small and inefficient.
+         * We adjust the batch sizes to never have a situation where a batch is smaller than (loadFactor * batchSizeHint).
+         */
+
+        float loadFactor = 0.1f;
```

Comment:
Should this be user-configurable?

Reply:
I think for now a sensible default is enough.